services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/allu/CableReportServiceAllu.kt

```diff
@@ -22,34 +23,38 @@ private val logger = KotlinLogging.logger {}
 class CableReportServiceAllu(
     private val webClient: WebClient,
     private val properties: AlluProperties,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO,
 ) : CableReportService {
 
     private val baseUrl = properties.baseUrl
 
-    private fun login(): String? {
+    private fun login(): String {
         try {
-            return webClient
-                .post()
-                .uri("$baseUrl/v2/login")
-                .contentType(MediaType.APPLICATION_JSON)
-                .accept(MediaType.TEXT_PLAIN)
-                .body(Mono.just(LoginInfo(properties.username, properties.password)))
-                .retrieve()
-                .bodyToMono(String::class.java)
-                .doOnError(WebClientResponseException::class.java) {
-                    logError("Error logging in to Allu", it)
-                }
-                // Allu has gone back and forth on whether it returns the login token surrounded
-                // with quotes or not. To be safe, remove them if they are found.
-                .map { it.trim('"') }
-                .block()
+            val token =
+                webClient
+                    .post()
+                    .uri("$baseUrl/v2/login")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.TEXT_PLAIN)
+                    .body(Mono.just(LoginInfo(properties.username, properties.password)))
+                    .retrieve()
+                    .bodyToMono(String::class.java)
+                    .doOnError(WebClientResponseException::class.java) {
+                        logError("Error logging in to Allu", it)
+                    }
+                    // Allu has gone back and forth on whether it returns the login token surrounded
+                    // with quotes or not. To be safe, remove them if they are found.
+                    .map { it.trim('"') }
+                    .block()
+
+            return token ?: throw AlluLoginException("Login token null")
         } catch (e: Throwable) {
-            throw AlluLoginException(e)
+            throw AlluLoginException(e.message)
```

Comment:
We want the complete exception here, not just the message. Otherwise, we'll lose valuable information on what caused the error.